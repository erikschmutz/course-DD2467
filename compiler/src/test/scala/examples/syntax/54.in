type 'a tree = Leaf | Node of 'a * 'a tree * 'a tree;;
let t = Node(4, Node(1, Leaf, Leaf), Leaf);;