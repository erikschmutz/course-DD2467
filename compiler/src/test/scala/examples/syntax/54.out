Program(List(TypeDeclaration(tree,TypeConstraints(List(TypeConstraint(Leaf,None),TypeConstraint(Node,Some(TypeTuple(List(TypeOption('a),TypeUsage(tree,List(TypeOption('a))),TypeUsage(tree,List(TypeOption('a))))))))),List(TypeOption('a))),Assignment(Identifier(t),Constructor(Node,Tuple(List(IntLit(4),Constructor(Node,Tuple(List(IntLit(1),Constructor(Leaf,Unit()),Constructor(Leaf,Unit())))),Constructor(Leaf,Unit())))))))